specification/monitor/resource-manager/Microsoft.Insights/stable/2021-05-01/metrics_API.json

```diff
@@ -0,0 +1,823 @@
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Insights/metrics": {
+      "get": {
+        "tags": [
+          "Metrics"
+        ],
+        "operationId": "Metrics_ListAtSubscriptionScope",
+        "description": "**Lists the metric data for a subscription**.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/RegionParameter"
+          },
+          {
+            "$ref": "#/parameters/TimespanParameter"
+          },
+          {
+            "$ref": "#/parameters/IntervalParameter"
+          },
+          {
+            "$ref": "#/parameters/MetricNamesParameter"
+          },
+          {
+            "$ref": "#/parameters/AggregationsParameter"
+          },
+          {
+            "$ref": "#/parameters/TopParameter"
+          },
+          {
+            "$ref": "#/parameters/OrderByParameter"
+          },
+          {
+            "$ref": "#/parameters/FilterParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionScopeResultTypeParameter"
+          },
+          {
+            "$ref": "#/parameters/MetricNamespaceParameter"
+          },
+          {
+            "$ref": "#/parameters/AutoAdjustTimegrain"
+          },
+          {
+            "$ref": "#/parameters/ValidateDimensions"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorContract"
+            }
+          },
+          "200": {
+            "description": "Successful request to get the list of metric values.",
+            "schema": {
+              "$ref": "#/definitions/SubscriptionScopeMetricResponse"
+            }
+          }
+        },
+        "x-ms-odata": "#/definitions/MetadataValue",
+        "x-ms-examples": {
+          "Get subscription level metric data": {
+            "$ref": "./examples/GetMultiResourceMetric.json"
+          },
+          "Get subscription level metric metadata": {
+            "$ref": "./examples/GetMultiResourceMetricMetadata.json"
+          }
+        }
+      },
+      "post": {
+        "tags": [
```

Comment:
What's the difference between GET and POST? Looks like there are returning the same result. Why do you need both? Usually POST is preferred to return metadata.

Reply:
Our existing metrics API just supports GET for fetching metric data so we wanted to keep that consistent, however because in this new API you can filter by resourceIds, you can quickly run into URL length issues so we also wanted to expose a POST version of the API to support filtering by a lot of resourceIds while avoiding URL length concerns.
If this isn't a common practice in ARM we could just go with the POST version, we just were trying to maintain some consistency with our existing API.

Reply:
I see. I think it's okay.